### PR TITLE
Add quiz logic and tables

### PR DESCRIPTION
**Pull request by lukecookssw**

### CODE REVIEW

File: API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/GetQuizDetailsQuery.cs [hunk 1]
```diff
@@ -34,69 +33,29 @@ public sealed class Handler : IRequestHandler<GetQuizDetailsQuery, QuizDetailsDt
...
-                        }
-                    }
-                };
+                return await _context.Quizzes
+                                        .Include(x => x.Questions)
+                                            .ThenInclude(x => x.Answers)
+                                        .Where(x => x.Id == request.QuizId)
+                                        .ProjectTo<QuizDetailsDto>(_mapper.ConfigurationProvider)
+                                        //.Select(x => new QuizDetailsDto
+                                        //{
```
Comment (matt-goldman) [inline API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/GetQuizDetailsQuery.cs:42]: Remove this unused code

File: API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/AutoMapperProfile.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace SSW.Rewards.Application.Quizzes.Queries.GetQuizDetailsQuery
+{
+    public class AutoMapperProfile : Profile
+    {
+        public AutoMapperProfile()
+        {
+            CreateMap<Domain.Entities.Quiz, QuizDetailsDto>()
+                .ForMember(dst => dst.QuizId, opt => opt.MapFrom(src => src.Id));
+            //.ForMember(dst => dst.Questions, opt => opt.MapFrom(src => src.Questions));
```
Comment (matt-goldman) [inline API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/AutoMapperProfile.cs:11]: Remove the unused lines from these profiles

**Review by matt-goldman — COMMENTED** [review]
Looks great! Some comments that got left in that need to be cleaned up.

File: API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/GetQuizDetailsQuery.cs (hunk 1 above)
Comment (lukecookssw) [reply API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/GetQuizDetailsQuery.cs]: Done

File: API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/AutoMapperProfile.cs (hunk 2 above)
Comment (lukecookssw) [reply API/SSW.Rewards.Application/Quizzes/Queries/GetQuizDetailsQuery/AutoMapperProfile.cs]: Done

**Review by matt-goldman — APPROVED** [review]
LGTM 🚀

**Pull request merged**